types/model/name.go

```diff
@@ -47,11 +54,7 @@ const (
 	PartBuild
 	PartDigest
 
-	// Invalid is a special part that is used to indicate that a part is
-	// invalid. It is not a valid part of a Name.
-	//
-	// It should be kept as the last part in the list.
-	PartInvalid
+	PartExtraneous = -1
```

Comment:
PartRest?

Reply:
Rest doesn't scream "Whoa this is unexpected!" as much.